klippy/gcode.py

```diff
@@ -156,19 +159,26 @@ def register_mux_command(self, cmd, key, value, func, desc=None):
         prev_values[value] = func
     def get_command_help(self):
         return dict(self.gcode_help)
+    def get_status(self, eventtime):
+        return self._get_status_info
+    def _build_status(self):
+        self._get_status_info['commands'] = sorted(
+            self.gcode_handlers.keys())
```

Comment:
The response dict needs to be fully rebuilt for proper change detection - that is: `self._get_status_info = {'commands': sorted(self.gcode_handlers.keys())}`

Reply:
I based the above on the existing code in [mcu.py](https://github.com/Klipper3d/klipper/blob/bb4711c5d31e8159945f945c662e6668059a174f/klippy/mcu.py); I assume that this `dict()` then negates proper change detection here?

https://github.com/Klipper3d/klipper/blob/bb4711c5d31e8159945f945c662e6668059a174f/klippy/mcu.py#L984-L985

Reply:
Yes, mcu.py looks okay to me, because it returns a new dict on every get_status() call.  You could also return a new dict() in gcode's get_status(), but rebuilding the dict in _build_status() is also sufficient.

-Kevin